client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoConnectionDataSource.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.jdbc;
+
+import io.trino.client.uri.PropertyName;
+
+import javax.sql.DataSource;
+
+import java.io.PrintWriter;
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.util.Properties;
+import java.util.logging.Logger;
+
+import static io.trino.client.uri.PropertyName.CATALOG;
+import static io.trino.client.uri.PropertyName.PASSWORD;
+import static io.trino.client.uri.PropertyName.SCHEMA;
+import static io.trino.client.uri.PropertyName.USER;
+
+public class TrinoConnectionDataSource
+        implements DataSource
+{
+    private String url;
+    private PrintWriter printWriter;
+
+    private final Properties connectionProperties = new Properties();
+    private final TrinoDriver trinoDriver;
+
+    public TrinoConnectionDataSource()
+    {
+        trinoDriver = new TrinoDriver();
+    }
+
+    @Override
+    public Connection getConnection()
+            throws SQLException
+    {
+        return getConnection(getUser(), getPassword());
+    }
+
+    @Override
+    public Connection getConnection(String user, String password)
+            throws SQLException
+    {
+        update(connectionProperties, USER, user);
+        update(connectionProperties, PASSWORD, password);
+
+        return trinoDriver.connect(url, connectionProperties);
+    }
+
+    @Override
+    public PrintWriter getLogWriter()
+            throws SQLException
+    {
+        return printWriter;
+    }
+
+    @Override
+    public void setLogWriter(PrintWriter out)
+            throws SQLException
+    {
+        this.printWriter = out;
+    }
+
+    @Override
+    public void setLoginTimeout(int seconds)
+            throws SQLException
+    {
+        DriverManager.setLoginTimeout(seconds);
+    }
+
+    @Override
+    public int getLoginTimeout()
+            throws SQLException
+    {
+        return DriverManager.getLoginTimeout();
+    }
+
+    @Override
+    public Logger getParentLogger()
+            throws SQLFeatureNotSupportedException
+    {
+        throw new SQLFeatureNotSupportedException("Parent logger not supported.");
```

Comment:
You can return a logger for `io.trino.jdbc`: `Logger.getLogger(TrinoDriver.class.getPackage().getName())`